### PR TITLE
initials for Travis CI config.

### PR DESCRIPTION
**Pull request by afred**

### CODE REVIEW

**Comment by afred** [comment]
Finally got Travis passing.
@randalldfloyd, @mccalluc, or @nianma, could one of you review and merge please? Or give me a :+1: and I'll do it. Thanks!

File: README.md [hunk 1]
```diff
@@ -1,3 +1,5 @@
+[![Build Status](https://travis-ci.org/projecthydra-labs/hydradam.svg?branch=HDM-37-set-up-Travis-CI-for-github-repo)](https://travis-ci.org/projecthydra-labs/hydradam)
```
Comment (mccalluc) [inline README.md]: @afred : should the branch reference be in here?

File: lib/fits.rb [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+      remove_fits!
+    end
+
+    private
+
+    def within_dir(dir)
+      orig_dir = Dir.pwd
+      Dir.chdir(dir)
+      yield if block_given?
+      Dir.chdir(orig_dir)
```
Comment (mccalluc) [inline lib/fits.rb]: @afred : actually, you can pass a block to chdir. We don't need to do this by hand. http://ruby-doc.org/core-2.2.0/Dir.html#method-c-chdir

File: lib/fits.rb [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+      run "chmod a+x #{script_name}"
+    end
+
+    def script_name
+      'fits.sh'
+    end
+
+    def make_symlink
+      Dir.chdir(unzipped_dir)
+      run "ln -s #{script_name} #{symlink}"
```
Comment (mccalluc) [inline lib/fits.rb]: Could be done without shell. http://ruby-doc.org/core-2.2.0/File.html#method-c-symlink

... which is probably gratuitous. Not a security problem here, but no telling where stuff gets copied and pasted to.

File: spec/spec_helper.rb [hunk 4]
```diff
@@ -10,6 +10,10 @@
 # in spec/support/ and its subdirectories.
 Dir[Rails.root.join("spec/support/**/*.rb")].each {|f| require f}
 
+# Global variable? Sure, why not? Use $in_travis in your tests to, say, exclude tests from running when on Travis CI.
+# NOTE: Then environment variable being checked here is set in .travis.yml.
+$in_travis = !ENV['TRAVIS'].nil? && ENV['TRAVIS'] == 'true'
+
```
Comment (mccalluc) [inline spec/spec_helper.rb]: This works, and probably gives us finer grained control, but another way is with rspec tags. ie, aapb .travis.yml:
```yaml
script: 'bundle exec rspec --tag ~not_on_travis'
```

advantage is that it makes it easier to emulate travis test run without tweaking environment variables.

File: lib/tasks/bawstun.rake [hunk 5]
```diff
@@ -1,6 +1,13 @@
...
+task :ci do
+
+  # Install fits
+  Rake::Task['fits:install'].invoke
+
+  # Download a clean copy of Jetty, preloaded with Solr and Fedora from the hydra-jetty gem.
+  Rake::Task['jetty:clean'].invoke
+
+  # Copy config from solr_conf/ and fedora_conf/ directories to Solr and Fedora downloaded from hydra-jetty gem.
   Rake::Task['jetty:config'].invoke
```
Comment (mccalluc) [inline lib/tasks/bawstun.rake]: Do we want to have a wait-until-jetty-is-running loop like on OV? Or is the start-up for this slow enough that we won't hit race conditions?

**Comment by mccalluc** [comment]
(But no show stoppers: It's great that you got this working!)

File: lib/fits.rb (hunk 2 above)
Comment (afred) [inline lib/fits.rb]: ha! nice, totally missed that bit of doc

File: spec/spec_helper.rb (hunk 4 above)
Comment (afred) [inline spec/spec_helper.rb]: meh. Environment variables aren't tweaked so much as simply set during the call to `rspec`, e.g.
```
# emulate travis
TRAVIS=true bundle exec rake spec
# don't emulate travis
bundle exec rake spec
```
i ripped off the `unless: $global_var` pattern from ScholarSphere. Hadn't seen it before and thought it was cool.

File: lib/tasks/bawstun.rake (hunk 5 above)
Comment (afred) [inline lib/tasks/bawstun.rake]: the wait happens just below... in a block passed to `Jettywrapper.wrap(jetty_params) do ...`

**Pull request merged**